plugins/default/src/Ide/Plugin/Pragmas.hs

```diff
@@ -67,27 +68,37 @@ codeActionProvider _ state _plId docId _ (J.CodeActionContext (J.List diags) _mo
     let mFile = docId ^. J.uri & uriToFilePath <&> toNormalizedFilePath'
     pm <- fmap join $ runAction "addPragma" state $ getParsedModule `traverse` mFile
     let dflags = ms_hspp_opts . pm_mod_summary <$> pm
-    -- Filter diagnostics that are from ghcmod
-        ghcDiags = filter (\d -> d ^. J.source == Just "typecheck") diags
+    -- Filter diagnostics that are from GHC
+        ghcDiags = filter isGhcDiag diags
     -- Get all potential Pragmas for all diagnostics.
         pragmas = concatMap (\d -> genPragma dflags (d ^. J.message)) ghcDiags
     cmds <- mapM mkCodeAction pragmas
     return $ Right $ List cmds
       where
+        isGhcDiag diag
+          | Just source <- diag ^. J.source
+          = source `elem` ["parser", "typecheck"]
```

Comment:
Are there any other sources that qualify as "GHC sources"? Things like: "renamer" and "simplifier"? I did a quick grep, but didn't find any.

Reply:
I think this covers them all, but it seems flimsy and I don't know why this filtering is needed at all. Perhaps try removing it and checking that nothing breaks?

Reply:
FYI, the other sources I have found by grepping are: "bios", "hlint", and "liquid". I don't know where/how to find an exhaustive list of them. I stuck to the filtering because the existing code was doing it, but I'll remove it.

By "checking that nothing breaks" I assume you mean: the tests still pass, right? :slightly_smiling_face:

Reply:
Correct